### PR TITLE
Feature/3 Added support for unifying versions

### LINKED ISSUE #3: Add support for unifying versions

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #3

### CODE REVIEW

File: src/main/java/com/exasol/artifactreferencechecker/ArtifactReferenceCheckerVerifyMojo.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                        "Found outdated artifact reference: " + matcher.group() + " in  " + this.currentFileName);
+                this.success = false;
+            }
+        }
+
+        @Override
+        public void report() throws MojoFailureException {
+            if (!this.success) {
+                throw new MojoFailureException(
+                        "Found invalid artifact references (see previous errors). You can use mvn artifact-reference-checker:unify to fix them.");
```
Comment (morazow) [inline src/main/java/com/exasol/artifactreferencechecker/ArtifactReferenceCheckerVerifyMojo.java]:
```suggestion
                        "Found invalid artifact references, please check the reported errors. You can use mvn artifact-reference-checker:unify to fix them.");
```

File: src/main/java/com/exasol/artifactreferencechecker/BaseReferenceCheckerMojo.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+     */
+    public BaseReferenceCheckerMojo(final FileAndLineVisitor fileAndLineVisitor, final Log logger) {
+        this.fileAndLineVisitor = fileAndLineVisitor;
+        this.logger = logger;
+    }
+
+    /**
+     * Iterate all files with correct file ending and their lines.
+     *
+     * @param project maven project (use to access the pom file)
```
Comment (morazow) [inline src/main/java/com/exasol/artifactreferencechecker/BaseReferenceCheckerMojo.java]:
```suggestion
     * @param project maven project used to access the pom file
```

File: src/main/java/com/exasol/artifactreferencechecker/BaseReferenceCheckerMojo.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        } catch (final IOException exception) {
+            throw new MojoExecutionException("Could not check files.", exception);
+        } catch (ExceptionWrapper exceptionWrapper) {
+            throw exceptionWrapper.getExecutionException();
+        }
+        fileVisitor.report();
+    }
+
+    /**
+     * File Visitor that checks if the files have the correct ending.
```
Comment (morazow) [inline src/main/java/com/exasol/artifactreferencechecker/BaseReferenceCheckerMojo.java]:
```suggestion
     * File Visitor that checks if the files have the correct extension.
```

File: src/main/java/com/exasol/artifactreferencechecker/BaseReferenceCheckerMojo.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+     * @param fileAndLineVisitor visit or unify specific behaviour
+     * @param logger             logger
+     */
+    public BaseReferenceCheckerMojo(final FileAndLineVisitor fileAndLineVisitor, final Log logger) {
+        this.fileAndLineVisitor = fileAndLineVisitor;
+        this.logger = logger;
+    }
+
+    /**
+     * Iterate all files with correct file ending and their lines.
```
Comment (morazow) [inline src/main/java/com/exasol/artifactreferencechecker/BaseReferenceCheckerMojo.java]:
```suggestion
     * Iterate all files with correct file extension and their lines.
```

File: src/main/java/com/exasol/artifactreferencechecker/ArtifactReferenceCheckerVerifyMojo.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import org.apache.maven.project.MavenProject;
+
+/**
+ * This class checks the artifacts references in a maven project.
+ */
+@Mojo(name = "verify", defaultPhase = LifecyclePhase.VERIFY)
+public class ArtifactReferenceCheckerVerifyMojo extends AbstractMojo {
+    private final BaseReferenceCheckerMojo baseReferenceCheckerMojo;
+    @Parameter(defaultValue = "${project}", required = true, readonly = true)
+    MavenProject project;
```
Comment (morazow) [inline src/main/java/com/exasol/artifactreferencechecker/ArtifactReferenceCheckerVerifyMojo.java]:
```suggestion
    @Parameter(defaultValue = "${project}", required = true, readonly = true)
    MavenProject project;
    
    private final BaseReferenceCheckerMojo baseReferenceCheckerMojo;
```

Comment (morazow) [reply src/main/java/com/exasol/artifactreferencechecker/ArtifactReferenceCheckerVerifyMojo.java]: Only to make it similar to `unify` counterpart.

**Review by morazow — CHANGES_REQUESTED** [review]

**Comment by morazow** [comment]
Did you also check with the local git repository? Please make sure that the git difference is only the modified lines by the plugin.

**Comment by jakobbraun** [comment]
> Did you also check with the local git repository? Please make sure that the git difference is only the modified lines by the plugin.

Yes. with dynamodb-virtual-schema

**Review by morazow — APPROVED** [review]
:+1:

**Pull request merged**